src/multiset.rs

```diff
@@ -157,28 +157,27 @@ impl<K> HashMultiSet<K> where
     /// use multiset::HashMultiSet;
     ///
     /// let mut multiset: HashMultiSet<i32> = HashMultiSet::new();
-    /// multiset.insert_times(5,3);
-    /// assert!(multiset.count_of(5) == 3);
-    /// multiset.remove_times(5,2);
-    /// assert!(multiset.count_of(5) == 1);
-    /// multiset.remove_times(5,1);
-    /// assert!(multiset.count_of(5) == 0);
-    /// multiset.remove_times(5,1);
-    /// assert!(multiset.count_of(5) == 0);
+    /// multiset.insert_times(5, 3);
+    /// assert!(multiset.count_of(&5) == 3);
+    /// assert!(multiset.remove_times(&5, 2));
+    /// assert!(multiset.count_of(&5) == 1);
+    /// assert!(multiset.remove_times(&5, 1));
+    /// assert!(multiset.count_of(&5) == 0);
+    /// assert!(!multiset.remove_times(&5, 1));
+    /// assert!(multiset.count_of(&5) == 0);
     /// ```
-    pub fn remove_times(&mut self, val: K, n: usize) {
-        match self.elem_counts.entry(val) {
-            Entry::Vacant(_) => (),
-            Entry::Occupied(mut view) => {
-                let v = *view.get();
-                if v > n {
-                    let v = view.get_mut();
-                    *v -= n
-                } else {
-                    let _ = view.remove_entry();
-                };
-            },
+    pub fn remove_times(&mut self, val: &K, times: usize) -> bool {
+        {
+            let entry = self.elem_counts.get_mut(val);
+            if entry.is_some() {
+                let count = entry.unwrap();
+                if *count > times {
+                    *count -= times;
+                    return true
+                }
+            }
         }
+        self.elem_counts.remove(val).is_some()
```

Comment:
Can you please revise this to use `match` with a `mut` reference? The last line is confusing until realizing that the case where only some items are removed returns early.

The cached `size` needs to be updated too, now that #12 is merged.

Reply:
I don't get this. What `match` on what `mut` reference?
Note that this statement needs to be outside of that extra scope above or the borrow checker will be complaining.